ambry-account/src/main/java/com/github/ambry/account/AbstractAccountService.java

```diff
@@ -151,19 +151,24 @@ public Collection<Container> updateContainers(String accountName, Collection<Con
       throw new AccountServiceException("Account " + accountName + " is not found", AccountServiceErrorCode.NotFound);
     }
 
+    if (containers.size() > Short.MAX_VALUE) {
+      accountServiceMetrics.maxContainersReachedForAccount.inc();
+      throw new AccountServiceException("Too many containers in request: " + containers.size(), AccountServiceErrorCode.BadRequest);
```

Comment:
QQ. Do we need this check? If containers.size() is greater than `Short.MAX_VALUE`, it would fail (or have failed) in the other condition?

Reply:
this is done to fail early if any request comes with more 32K containers. if you think this is unnecessary, i can remove it